verifier/providerverifier.go

```diff
@@ -0,0 +1,280 @@
+package verifier
+
+import (
+	"context"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/openpubkey/openpubkey/gq"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+	oidcclient "github.com/zitadel/oidc/v2/pkg/client"
+)
+
+type ProviderVerifier struct {
+	issuer          string
+	commitmentClaim string
+	options         ProviderVerifierOpts
+}
+
+type ProviderVerifierOpts struct {
+	// If ClientID is specified, then verification will require that the ClientID
+	// be present in the audience ("aud") claim of the PK token payload
+	ClientID string
```

Comment:
Given the how common and how damaging failure to check `aud=client-id` has been in the OIDC ecosystem, we should not default to not checking the `aud=client-id`. For particular OPs like Google the audience should be required and we should not allow a developer to turn it off without writing a custom OP.

Reply:
[0d9c454](https://github.com/openpubkey/openpubkey/pull/88/commits/0d9c454585fe3122a692e2ead0e6031f100b8bb0)